nestkernel/mpi_manager.cpp

```diff
@@ -87,7 +87,19 @@ nest::MPIManager::init_mpi( int* argc, char** argv[] )
     comm = kernel().music_manager.communicator();
 #else  /* #ifdef HAVE_MUSIC */
     int provided_thread_level;
-    MPI_Init_thread( argc, argv, MPI_THREAD_FUNNELED, &provided_thread_level );
+    int mpi_retcode, mpi_error_len;
+    char* mpi_error_string;
+
+    mpi_retcode = MPI_Init_thread(
+      argc, argv, MPI_THREAD_FUNNELED, &provided_thread_level );
+    if ( mpi_retcode != 0 )
+    {
+      MPI_Error_string( mpi_retcode, mpi_error_string, &mpi_error_len );
+      std::string msg = String::compose(
+        "Error initializing MPI. Error encountered: %1", mpi_error_string );
+
+      LOG( M_ERROR, "MPIManager::init_mpi", msg );
```

Comment:
I'm not entirely sure. This seems to be the only way of getting information from a failed MPI_Init_thread call - or the only way I've found in the documentation. I'm running a simulation now that will hopefully fail and confirm if this works or not. Unfortunately, it's still stuck in the cluster queue.

The page [here](http://www.mpich.org/static/docs/v3.1/www3/MPI_Init_thread.html) says:

```
Errors

All MPI routines (except MPI_Wtime and MPI_Wtick) return an error value; C routines as the 
value of the function and Fortran routines in the last argument. Before the value is returned, 
the current MPI error handler is called. By default, this error handler aborts the MPI job. The 
error handler may be changed with MPI_Comm_set_errhandler (for communicators), 
MPI_File_set_errhandler (for files), and MPI_Win_set_errhandler (for RMA windows). The 
MPI-1 routine MPI_Errhandler_set may be used but its use is deprecated. The predefined 
error handler MPI_ERRORS_RETURN may be used to cause error values to be returned. 
Note that MPI does not guarentee that an MPI program can continue past an error; however, 
MPI implementations will attempt to continue whenever possible.

MPI_SUCCESS
    No error; MPI routine completed successfully. 

MPI_ERR_OTHER
    Other error; use MPI_Error_string to get more information about this error code. 
```

Note that it does mention setting an error handler - maybe that would be a better way of going about it - setting default error handlers right at the beginning? 

I also just noticed that `MPI_ERRORS_RETURN` needs to be set to get the method to return an error value. I'll look at how this is to be done and update the code.